keras_nlp/models/gpt_neox/gpt_neox_backbone_test.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test for GPTNeoX backbone models."""
+
+import os
+
+import pytest
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_nlp.models.gpt_neox.gpt_neox_backbone import GPTNeoXBackbone
+
+
+class GPTNeoXTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        # For DTensor.
+        keras.backend.experimental.enable_tf_random_generator()
+        keras.utils.set_random_seed(1337)
+
+        self.backbone = GPTNeoXBackbone(
+            vocabulary_size=10,
+            num_layers=4,
+            num_heads=4,
+            hidden_dim=64,
+            intermediate_dim=64,
+            max_sequence_length=10,
+        )
+        self.input_batch = {
+            "token_ids": tf.ones((2, 5), dtype="int32"),
+            "padding_mask": tf.ones((2, 5), dtype="int32"),
+        }
+        self.input_dataset = tf.data.Dataset.from_tensor_slices(
+            self.input_batch
+        ).batch(2)
+
+    def test_call(self):
+        self.backbone(self.input_batch)
+
+    def test_token_embedding(self):
+        output = self.backbone.token_embedding(self.input_batch["token_ids"])
+        self.assertEqual(output.shape, (2, 5, 64))
+
+    # def test_name(self):
```

Comment:
@mattdangerw this test is failing. How to fix that?

Reply:
Do you know what the actual value of `self.backbone.name` is? I wonder if some case conversion code is getting messed up.

Reply:
no clue here.

Reply:
When you run the test, it should give you an error that tell your value x != value y right? What is the test failure message?

Reply:
![image](https://github.com/keras-team/keras-nlp/assets/51750587/6bcc5613-73f9-4454-b626-89d47e0a9623)

why is there an `_1` at last?

Reply:
The `_1` would be from creating two backbones with the default name in the same session. I am able to get this test passing simply by uncommenting it and changing `"gpt_neox_backbone"` to `"gpt_neo_x_backbone"`.

Which actually brings up a good points. We should actually rename the directly and all files from `gpt_neox_...` to -> `gpt_neo_x_...` to match the way Keras automatically converts Camel to snake_case.

Reply:
I think this is still unresolved. We should rename this directory and all files to `gpt_neo_x...` and then enable this test. It should pass at that point.